sklearn/utils/estimator_checks.py

```diff
@@ -1492,6 +1492,21 @@ def check_no_fit_attributes_set_in_init(name, Estimator):
                 'estimator but in the fit method. Attribute {!r} '
                 'was found in estimator {}'.format(attr, name))
 
+    if(Estimator.__name__ not in ["GaussianProcess", "RandomizedLasso",
+                                  "RandomizedLogisticRegression",
+                                  "RandomizedPCA"] and
+       sys.version_info > (3, 5)):
+        # This check is only for non-decorated (eg: deprecated) estimator
+        # _get_args and _get_parent_args only work for python 3.5
+        init_params = _get_args(estimator.__init__)
+        base_params = (_get_parent_args(Estimator) +
+                       [a for a in dir(Estimator)
+                        if not a.startswith("__")])
+        for attr, val in vars(estimator).items():
+            if attr not in base_params:
+                assert_true(attr in init_params)
```

Comment:
The current message in case of error is `AssertionError: False is not true`, and it's not easy to understand what is the issue. We should give a `msg` to `assert_true`, with at least the names of the estimator and of the attribute.